src/collection.js

```diff
@@ -57,13 +57,30 @@ export default class Collection {
 		this._idProperty = options && options.idProperty || 'id';
 
 		/**
-		 * A helper mapping external items from bound collection ({@link #bindTo})
-		 * and actual items of the collection.
+		 * A helper mapping external items of a bound collection ({@link #bindTo})
+		 * and actual items of this collection.
 		 *
 		 * @protected
 		 * @member {Map}
 		 */
-		this._boundItemsMap = new Map();
+		this._bindToExternalToInternalMap = new Map();
+
+		/**
+		 * A helper mapping items of this collection to external items of a bound collection
+		 * ({@link #bindTo}).
+		 *
+		 * @protected
+		 * @member {Map}
+		 */
+		this._bindToInternalToExternalMap = new Map();
```

Comment:
Can't it be a `WeaMap`? Does not look like a real collection more like mapping and `WeaMap` wound be safer for mapping because it will be cleared even if one forget to call `clear`. I see you iterate over this map only in tests what seems to be too deeply testing, checking internal, what is a bad smell. Can't test check only the content of collections without checking internal?